sw/device/silicon_creator/manuf/lib/provisioning.c

```diff
@@ -169,27 +213,60 @@ flash_ctrl_owner_secret_write(dif_flash_ctrl_state_t *flash_state) {
 }
 
 /**
- * Configures the Silicon Creator seed secret shares in the SECRET2 OTP
- * partition.
+ * Wraps an RMA unlock token by encrypting it with an RSA-3072 public key.
+ *
+ * @param rma_token RMA unlock token to encrypt.
+ * @param[out] encrypted_rma_token Encrypted RMA unlock token.
+ * @return OK_STATUS on success.
+ */
+OT_WARN_UNUSED_RESULT status_t wrap_rma_unlock_token(
+    const rsa_3072_public_key_t *key, const uint64_t *rma_token,
+    rsa_3072_int_t *encrypted_rma_token) {
+  // Compute Montgomery constants.
+  rsa_3072_constants_t constants;
+  TRY(rsa_3072_compute_constants(key, &constants));
+
+  // Convert RMA unlock token to RSA 3072 int.
+  rsa_3072_int_t formatted_rma_token;
+
+  // TODO(#17393): add padding?
```

Comment:
@jadephilipoom: since the underlying data being encrypted here (i.e., the RMA unlock token) is only 128 bits and generated by the CSRNG (i.e., is random data), could we just pad out with more random data, as follows: <random bits><RMA unlock token>, then on the host side just drop the leading <random bits> when we decrypt? Or should we still use one of the planned supported [padding schemes](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/crypto/include/rsa.h;drc=aab08735df9372ee5c7010e516dbe11851e9c7af;l=29) in cryptolib?

Reply:
For RSA, the padding choice is actually a security-critical parameter\*, and RSA is not safe to use in general unless the data is padded. I think we should use one of the planned padding schemes, although as I said in #18126 I'd lean towards avoiding RSA encryption altogether if we can.

\* For example, the attack from section 4.4 in this [RSA attack survey](https://crypto.stanford.edu/~dabo/pubs/papers/RSA-survey.pdf).